includes/server/class-llms-rest-memberships-controller.php

```diff
@@ -0,0 +1,590 @@
+				'lifterlms' ),
+			'type'        => 'integer',
+			'context'     => array( 'view', 'edit' ),
+			'arg_options' => array(
+				'sanitize_callback' => 'absint',
+			),
+		);
+		$schema['properties']['restriction_url']     = array(
+			'description' => __(
+				'URL used for redirecting non-members when `restriction_action` is `custom`.',
+				'lifterlms' ),
+			'type'        => 'string',
+			'context'     => array( 'view', 'edit' ),
+			'format'      => 'uri',
+			'arg_options' => array(
+				'sanitize_callback' => 'esc_url_raw',
+			),
+		);
+		$schema['properties']['sales_page_page_id']  = array(
+			'description' => __(
+				'The WordPress page ID of the sales page. Required when `sales_page_type` equals `page`. ' .
+				'Only returned when the `sales_page_type` equals `page`.',
+				'lifterlms' ),
+			'type'        => 'integer',
+			'context'     => array( 'view', 'edit' ),
+			'arg_options' => array(
+				'sanitize_callback' => 'absint',
+			),
+		);
+		$schema['properties']['sales_page_type']     = array(
+			'description' => __(
+				'Defines alternate content displayed to visitors and non-enrolled students when accessing the post. ' .
+				'- `none` displays the post content. ' .
+				'- `content` displays alternate content from the `excerpt` property.' .
+				'- `page` redirects to the WordPress page defined in `content_page_id`.' .
+				'- `url` redirects to the URL defined in `content_page_url`.',
+				'lifterlms' ),
+			'type'        => 'string',
+			'default'     => 'none',
+			'enum'        => array_keys( llms_get_sales_page_types() ),
+			'context'     => array( 'view', 'edit' ),
+		);
+		$schema['properties']['sales_page_url']      = array(
+			'description' => __(
+				'The URL of the sales page content. Required when `sales_page_type` equals `url`. ' .
+				'Only returned when the `sales_page_type` equals `url`.',
+				'lifterlms' ),
+			'type'        => 'string',
+			'context'     => array( 'view', 'edit' ),
+			'format'      => 'uri',
+			'arg_options' => array(
+				'sanitize_callback' => 'esc_url_raw',
+			),
+		);
+		$schema['properties']['tags']                = array(
+			'description' => __( 'List of membership tags.', 'lifterlms' ),
+			'type'        => 'array',
+			'items'       => array(
+				'type' => 'integer',
+			),
+			'context'     => array( 'view', 'edit' ),
+		);
+
+		return $schema;
+	}
+
+	/**
+	 * Get object.
+	 *
+	 * @since [version]
+	 *
+	 * @param int $id Object ID.
+	 * @return LLMS_Membership|WP_Error
+	 */
+	protected function get_object( $id ) {
+		/** @var LLMS_Membership $membership */
+		$membership = llms_get_post( $id );
+
+		return is_a( $membership, 'LLMS_Membership' ) ? $membership : llms_rest_not_found_error();
+	}
+
+	/**
+	 * Retrieve an ID from the object.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Membership $object LLMS_Membership object.
+	 * @return int
+	 */
+	protected function get_object_id( $object ) {
```

Comment:
This method is not needed.
It was "needed" in the Courses controller because the `LLMS_Course` has the `get_id()` method which is [deprecated](https://github.com/gocodebox/lifterlms/blob/3.36.2/includes/models/model.llms.course.php#L806), and according to `get_object_id()` defined in `LLMS_REST_Controller_Stubs` takes precedence over `get('id')`
https://github.com/gocodebox/lifterlms-rest/blob/1.0.0-beta.7/includes/abstracts/class-llms-rest-controller-stubs.php#L107
LLMS_Membership has no `get_id()` method, you can remove it then :)